provider/aws/aws.go

```diff
@@ -926,6 +961,39 @@ func (p *AWSProvider) newChange(action route53types.ChangeAction, ep *endpoint.E
 		if useGeolocation {
 			change.ResourceRecordSet.GeoLocation = geolocation
 		}
+
+		geoproximity := &route53types.GeoProximityLocation{}
+		useGeoproximity := false
+		if prop, ok := ep.GetProviderSpecificProperty(providerSpecificGeoProximityLocationAWSRegion); ok {
```

Comment:
Move to own method, do not bloat current pls

Reply:
done, thanks